File: src/main/java/compiler/block/loops/ForBlock.java
package compiler.block.loops;

import compiler.Parameter;
import compiler.block.Block;

public class ForBlock extends Block {

    private String type = "for";
    private Parameter[] params;
    String name;

    public ForBlock(Block superBlock, String name) {
        super(superBlock, true, false);
        this.name = name;

    }

    public String getName() {
        return name;
    }

    @Override
    public String getValue() {
        return null;
    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return null;
    }

    @Override
    public String getClosingCode() {
        return null;
    }

    @Override
    public String getBodyCode() {
        return null;
    }

    public Parameter[] getParameters() {
        return params;
    }


    @Override
    public void init() {

    }

    @Override
    public void run() {

    }
}

File: src/main/scala/compiler/block/loops/ForBlock.scala
package compiler.block.loops

import compiler.block.Block
import compiler.structure.parameters.Parameter

/**
  * Represents a for loop
  *
  * @param superBlockInit
  * @param name
  */
class ForBlock(var superBlockInit: Block, var name: String) extends Block(superBlockInit, true, false) {

  private val params: Array[Parameter] = null

  def getParameters: Array[Parameter] = {
    return params
  }

  def init() {
  }

  def getName: String = name

  def getValue: String = ""

  def getType: String = ""

  def getOpeningCode: String = ""

  def getClosingCode: String = ""

  override def toString: String = "for: " + name

}